ui/components/MesheryPlayComponent.js

```diff
@@ -83,211 +83,184 @@ const styles = (theme) => ({
   },
 });
 
-class MesheryPlayComponent extends React.Component {
-  constructor(props) {
-    super(props);
+const MesheryPlayComponent = (props) => {
+  const router = useRouter();
+  const { meshAdapters, classes } = props;
+  const [adapter, setAdapterState] = useState({});
 
-    const { meshAdapters } = props;
-    let adapter = {};
-    if (meshAdapters && meshAdapters.size > 0) {
-      adapter = meshAdapters[0];
-    }
-    this.state = {
-      adapter,
-    };
-  }
-
-  handleRouteChange = () => {
-    const queryParam = this.props?.router?.query?.adapter;
+  const handleRouteChange = useCallback(() => {
+    const queryParam = router.query.adapter;
     if (queryParam) {
-      const selectedAdapter = this.props.meshAdapters.find(
+      const selectedAdapter = meshAdapters.find(
         ({ adapter_location }) => adapter_location === queryParam,
       );
       if (selectedAdapter) {
-        this.setState({ adapter: selectedAdapter });
+        setAdapterState(selectedAdapter);
```

Comment:
Are u sure this state got updated?

https://react.dev/learn/updating-objects-in-state

Reply:
Sorry My bad. Let me make some changes.

Reply:
Done the changes. Somehow the commit made by l5io is also included in the commit that I made.